pkg/resource/bucket/sdk.go

```diff
@@ -154,11 +154,28 @@ func (rm *resourceManager) sdkCreate(
 	if err != nil {
 		return nil, err
 	}
-	if input.CreateBucketConfiguration == nil {
-		input.CreateBucketConfiguration = &svcsdktypes.CreateBucketConfiguration{}
-	}
-	if input.CreateBucketConfiguration.LocationConstraint == "" && rm.awsRegion != "us-east-1" {
-		input.CreateBucketConfiguration.LocationConstraint = svcsdktypes.BucketLocationConstraint(rm.awsRegion)
+	// Only set CreateBucketConfiguration if we're not in us-east-1
+	// and need to specify a different region
+	if rm.awsRegion != "us-east-1" {
+		// Check if user specified a different region than the controller's region
+		if desired.ko.Spec.CreateBucketConfiguration != nil &&
+			desired.ko.Spec.CreateBucketConfiguration.LocationConstraint != nil {
+			userRegion := *desired.ko.Spec.CreateBucketConfiguration.LocationConstraint
+			if userRegion != string(rm.awsRegion) {
+				msg := fmt.Sprintf("LocationConstraint must be %s but got %s", rm.awsRegion, userRegion)
+				reason := "InvalidLocationConstraint"
+				ackcondition.SetSynced(desired, corev1.ConditionFalse, &msg, &reason)
+				ackcondition.SetTerminal(desired, corev1.ConditionTrue, &msg, &reason)
+				return desired, ackerr.Terminal
```

Comment:
You can just return a terminal error, the reconcile logic in runtime will take care of injecting it in the conditions.